packages/cardano-services/src/StakePool/HttpStakePoolMetadata/HttpStakePoolMetadataService.ts

```diff
@@ -93,6 +93,7 @@ export const createHttpStakePoolMetadataService = (
           return { errors: [new CustomError('Invalid metadata: Missing ext signature or public key')], metadata };
         }
 
+        // TODO: We probably need a delay here
```

Comment:
I haven't seen this behavior when testing it manually with a few pool metadata URLs.
So the case is when the first metadata [fetch](https://github.com/input-output-hk/cardano-js-sdk/pull/640/files#diff-a125636e823a5a5ec5840f90b0ce619515a214b5f0523bdd41db32eaa011c1f4R77) has finished, the [second](https://github.com/input-output-hk/cardano-js-sdk/pull/640/files#diff-a125636e823a5a5ec5840f90b0ce619515a214b5f0523bdd41db32eaa011c1f4R97) one sometimes fails with what error? Is it reproducible with the same URLs always?
If it depends on the external pool metadata servers and the delay helps, not sure what other approach we can take atm either. @AngelCastilloB

Reply:
We need to be resilient to network outages too, so ideally we would make an immediate attempt, then if it fails, retry a few times before giving up for this job anyway.

Reply:
Yes, for the first we can use the same `pRetry` used by [createDnsResolver](https://github.com/input-output-hk/cardano-js-sdk/blob/23bdb6ebb664da90ae5bf877d7a631e13eedfacf/packages/cardano-services/src/Program/utils.ts#L35-L40).

Reply:
We should push the retry logic to the worker, because that will be needed either way. The fetcher just needs to return more granular error types (either with some enum like with ProvideFailure, or just distinct error types), so that it can decide what to do.

Reply:
The error granularity looks good to me👍🏼